FESS-GUI/mainwindow.cpp

```diff
@@ -522,19 +522,160 @@ void MainWindow::on_lineEditPassword_textEdited(const QString &password)
+void MainWindow::runFlywheelOperations()
+{
+    flywheelOperation->sync();
+}
+
+void MainWindow::startFlywheelInterface()
+{
+    deviceInterface = interfaceManager->getCurrentInterface();
+
+    if(!deviceInterface)
+    {
+        errorInterafceNotDefined();
+    }
+    else
+    {
+        deviceInterface->startDevice();
+        flywheelOperation->setInterface(deviceInterface);
+        flywheelRefreshTimer->start();
+        ui->outputLog->append(QString("%1: Interface started: %2").arg(QTime::currentTime().toString(),deviceInterface->name()));
+    }
+}
+
+void MainWindow::stopFlywheelInterface()
+{
+    if(!deviceInterface)
+    {
+        errorInterafceNotDefined();
+    }
+    else
+    {
+        flywheelRefreshTimer->stop();
+        deviceInterface->stopDevice();
+        ui->outputLog->append(QString("%1: Interface stopped: %2").arg(QTime::currentTime().toString(),deviceInterface->name()));
+    }
+}
+
+void MainWindow::closeFlywheelInterface()
+{
+    if(!deviceInterface)
+    {
+        errorInterafceNotDefined();
+    }
+    else
+    {
+        flywheelRefreshTimer->stop();
+        deviceInterface->stopDevice();
+
+        ui->actionDeviceIndicator->setText(QString("Device: None"));
+        ui->outputLog->append(QString("%1: Interface stopped: %2").arg(QTime::currentTime().toString(),deviceInterface->name()));
+        ui->outputLog->append(QString("%1: Interface closed: %2").arg(QTime::currentTime().toString(),deviceInterface->name()));
+
+        interfaceManager->closeCurrentInterface();
+        deviceInterface = interfaceManager->getCurrentInterface();
+    }
+}
+
+// Dialog Windows
+
+void MainWindow::openInterfaceSettingsWindow(){  //show the password dialog box
+    CommonInterfaceSelector* deviceSettingsWindow = new CommonInterfaceSelector(interfaceManager, this);
+    connect(deviceSettingsWindow, SIGNAL(finished(int)), this, SLOT(closeInterfaceSettingsWindow()));
+    deviceSettingsWindow->show();
+}
+
+void MainWindow::closeInterfaceSettingsWindow()
+{
+    deviceInterface = interfaceManager->getCurrentInterface();
+
+    if (deviceInterface)
+    {
+        ui->actionDeviceIndicator->setText(QString("Device: %1").arg(deviceInterface->name()));
+        ui->outputLog->append(QString("%1: Interface set to: %2").arg(QTime::currentTime().toString(),deviceInterface->name()));
+
+        deviceInterface->startDevice();
+        ui->outputLog->append(QString("%1: Interface started: %2").arg(QTime::currentTime().toString(),deviceInterface->name()));
+
+        flywheelOperation->setInterface(deviceInterface);
+        flywheelRefreshTimer->start();
+    }
+    else
+    {
+        ui->actionDeviceIndicator->setText(QString("Device: None"));
+    }
+}
+
+void MainWindow::on_actionSet_Reset_Password_triggered() //show the password dialog box
+{
+    SetPasswordDialog* passwordResetWindow = new SetPasswordDialog(this);
+    passwordResetWindow->show();
+}
+
+// Signal Setups
+
+void MainWindow::setUpSignals()
```

Comment:
Move these to right after the constructor, so all constructor code is in one place